src/Spice86.Aeon/Emulator/Video/Dac.cs

```diff
@@ -70,70 +64,66 @@ public byte WriteIndex
         /// <returns>Red, green, or blue channel value.</returns>
         public byte Read()
         {
-            unsafe
+            Rgb color = _palette[_readIndex];
+            _readChannel++;
+            switch (_readChannel)
             {
-                uint color = palette[readIndex];
-                readChannel++;
-                if (readChannel == 1)
-                {
-                    return (byte)((color >> 18) & 0x3F);
-                }
-
-                if (readChannel == 2)
-                {
-                    return (byte)((color >> 10) & 0x3F);
-                }
-
-                readChannel = 0;
-                readIndex++;
-                return (byte)((color >> 2) & 0x3F);
+                case 1:
+                    return color.G;
+                case 2:
+                    return color.B;
             }
+
+            _readChannel = 0;
+            _readIndex++;
+            return color.R;
         }
+        
         /// <summary>
         /// Writes the next channel in the current color.
         /// </summary>
         /// <param name="value">Red, green, or blue channel value.</param>
-        public void Write(byte value)
-        {
-            unsafe
+        public void Write(byte value) {
+            _writeChannel++;
+            Rgb color = _palette[_writeIndex];
+            switch (_writeChannel)
             {
-                writeChannel++;
-                if (writeChannel == 1)
-                {
-                    palette[writeIndex] &= 0xFF00FFFF;
-                    palette[writeIndex] |= (uint)((value & 0x3F) << 18);
-                }
-                else if (writeChannel == 2)
-                {
-                    palette[writeIndex] &= 0xFFFF00FF;
-                    palette[writeIndex] |= (uint)((value & 0x3F) << 10);
-                }
-                else
-                {
-                    palette[writeIndex] &= 0xFFFFFF00;
-                    palette[writeIndex] |= (uint)((value & 0x3F) << 2);
-                    writeChannel = 0;
-                    writeIndex++;
-                }
+                // value * 255 / 63, or else colors are way too dark on screen
+                // We could shift by 2 instead, but while it's faster,
+                // it may not be as accurate.
+                case 1:
+                    color.G = _lightCorrectedValuesLookupTable[value];
+                    break;
+                case 2:
+                    color.B = _lightCorrectedValuesLookupTable[value];
+                    break;
+                default:
+                    color.R = _lightCorrectedValuesLookupTable[value];
```

Comment:
```suggestion
                    color.R = _lightCorrectedValuesLookupTable[value];
                    break;
                case 2:
                    color.G = _lightCorrectedValuesLookupTable[value];
                    break;
                default:
                    color.B = _lightCorrectedValuesLookupTable[value];
```
_writeChannel is increased before being used, so 1 is the first color (R)